test/e2e/misc_test.go

```diff
@@ -76,6 +88,135 @@ func (suite *MiscTestSuite) TestValidateBuildImage() {
 	}
 }
 
+func (suite *MiscTestSuite) TestBasicOAuth() {
+	if !isOpenShift(t) {
+		t.Skip("This test only runs on Openshift")
+	}
+
+	username := "e2etestuser"
+	password := "befuddled"
+	// To update this, use the output from: htpasswd -nbs e2etestuser befuddled
+	userPasswordHash := "e2etestuser:{SHA}MfHY85GCR7WcTE7cQ2CGmXg9uTA="
+
+	userPasswordSecret := make(map[string][]byte)
+	userPasswordSecret["htpasswd"] = []byte(userPasswordHash)
+	secret := createSecret("test-oauth-secret", namespace, userPasswordSecret)
+
+	jaegerInstanceName := "test-oauth"
+	jaeger := jaegerWithPassword(namespace, jaegerInstanceName, secret.Name)
+	err := fw.Client.Create(context.Background(), jaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying jaeger")
+	defer undeployJaegerInstance(jaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, jaegerInstanceName, 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for Jaeger deployment")
+
+	urlPattern := "%s/api/services"
+	route := findRoute(t, fw, jaegerInstanceName)
+	require.Len(t, route.Status.Ingress, 1, "Wrong number of ingresses.")
+	url := fmt.Sprintf("https://"+urlPattern, route.Spec.Host)
+	transport := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	httpClient := http.Client{Timeout: 30 * time.Second, Transport: transport}
+	log.Infof("Using Query URL [%v]", url)
+
+	// A request without credentials should return a 403
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	require.NoError(t, err, "Failed to create httpRequest")
+	response, err := httpClient.Do(request)
+	require.NoError(t, err)
+	require.Equal(t, response.StatusCode, 403)
+
+	// Add basic auth to the request and retry
+	request.SetBasicAuth(username, password)
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		response, err := httpClient.Do(request)
+		require.NoError(t, err)
```

Comment:
Would still like to see a check for status code 200 - just to re-enforce fact that authentication has now happened, even though getting a service list should imply that - or do you feel that is overkill?

Reply:
I added the check.  It's probably not strictly needed, but will be useful for resolving test failures.